test/common_test/simulation_features.cc

```diff
@@ -520,31 +538,77 @@ TYPED_TEST(SimulationFeaturesTestBasic, FreeGroup)
     auto freeGroupLink = link->FindFreeGroup();
     ASSERT_NE(nullptr, freeGroupLink);
 
-    StepWorld<Features>(world, true);
+    StepWorld<FreeGroupFeatures>(world, true);
 
+    // Set initial pose.
+    const gz::math::Pose3d initialPose{0, 0, 2, 0, 0, 0};
     freeGroup->SetWorldPose(
-      gz::math::eigen3::convert(
-        gz::math::Pose3d(0, 0, 2, 0, 0, 0)));
-    freeGroup->SetWorldLinearVelocity(
-      gz::math::eigen3::convert(gz::math::Vector3d(0.1, 0.2, 0.3)));
-    freeGroup->SetWorldAngularVelocity(
-      gz::math::eigen3::convert(gz::math::Vector3d(0.4, 0.5, 0.6)));
-
-    auto frameData = model->GetLink(0)->FrameDataRelativeToWorld();
-    EXPECT_EQ(gz::math::Pose3d(0, 0, 2, 0, 0, 0),
-              gz::math::eigen3::convert(frameData.pose));
+      gz::math::eigen3::convert(initialPose));
+
+    // Set initial velocities.
+    const Eigen::Vector3d initialLinearVelocity{0.1, 0.2, 0.3};
+    const Eigen::Vector3d initialAngularVelocity{0.4, 0.5, 0.6};
+    freeGroup->SetWorldLinearVelocity(initialLinearVelocity);
+    freeGroup->SetWorldAngularVelocity(initialAngularVelocity);
+
+    auto freeGroupFrameData = freeGroup->FrameDataRelativeToWorld();
+    auto linkFrameData = model->GetLink(0)->FrameDataRelativeToWorld();
+
+    // Before stepping, check that pose matches what was set.
+    EXPECT_EQ(initialPose,
+              gz::math::eigen3::convert(freeGroupFrameData.pose));
+    EXPECT_EQ(initialPose,
+              gz::math::eigen3::convert(linkFrameData.pose));
+
+    // Before stepping, check that velocities match what was set.
+    AssertVectorApprox vectorPredicateVelocity(1e-7);
+    EXPECT_PRED_FORMAT2(vectorPredicateVelocity,
+      initialLinearVelocity,
+      freeGroupFrameData.linearVelocity);
+    EXPECT_PRED_FORMAT2(vectorPredicateVelocity,
+      initialLinearVelocity,
+      linkFrameData.linearVelocity);
+    EXPECT_PRED_FORMAT2(vectorPredicateVelocity,
+      initialAngularVelocity,
+      freeGroupFrameData.angularVelocity);
+    EXPECT_PRED_FORMAT2(vectorPredicateVelocity,
+      initialAngularVelocity,
+      linkFrameData.angularVelocity);
 
     // Step the world
-    StepWorld<Features>(world, false);
-    // Check that the first link's velocities are updated
-    frameData = model->GetLink(0)->FrameDataRelativeToWorld();
-    EXPECT_TRUE(gz::math::Vector3d(0.1, 0.2, 0.3).Equal(
-                gz::math::eigen3::convert(frameData.linearVelocity), 0.1));
-    EXPECT_EQ(gz::math::Vector3d(0.4, 0.5, 0.6),
-              gz::math::eigen3::convert(frameData.angularVelocity));
+    StepWorld<FreeGroupFeatures>(world, false);
+
+    // Check the velocities again.
+    freeGroupFrameData = freeGroup->FrameDataRelativeToWorld();
+    linkFrameData = model->GetLink(0)->FrameDataRelativeToWorld();
+
+    // Expect linear velocity to be affected by gravity.
+    const Eigen::Vector3d linearVelocityAfterStep{0.1, 0.2, 0.3 - 9.8 * 0.001};
```

Comment:
TPE is failing these expectations

Reply:
should be fixed by https://github.com/gazebosim/gz-physics/pull/567/commits/d604c92e83d587c26c67a5c9d6a37c3465961603